src/ralph/export_to_ng/subscribers.py

```diff
@@ -228,6 +229,47 @@ def sync_rack_to_ralph2(data):
         publish_sync_ack_to_ralph3(rack, data['id'])
 
 
+def _get_or_create_obj(model, data, ralph_id_key='ralph2_id'):
+    created = False
+    if data[ralph_id_key]:
+        obj = model.objects.get(pk=data[ralph_id_key])
+    else:
+        created = True
+        obj = model()
+    return obj, created
+
+
+@sync_subscriber(
+    topic='sync_virtual_server_to_ralph2',
+    disable_publishers=[sync_device_to_ralph3]
+)
+def sync_virtual_server_to_ralph2(data):
+    vs, created = _get_or_create_obj(Device, data)
+
+    vs.hostname = data['hostname']
+    vs.sn = data['sn']
+    model, _ = DeviceModel.objects.get_or_create(
+        type=DeviceType.virtual_server, name=data['type']
+    )
+    vs.model = model
+    if data['service_uid']:
+        vs.service = ServiceCatalog.objects.get(uid=data['service_uid'])
+    else:
+        vs.service = None
+    vs.device_environment_id = data['environment_id']
+    if data['venture_id'] and data['venture_role_id']:
+        vs.venture_id = data['venture_id']
+        vs.venture_role_id = data['venture_role_id']
+    else:
+        logger.error('Venture role is None for Device with id {} (virtual server)'.format(  # noqa
```

Comment:
it's definately not error (venture is not required). maybe warning, or even info?